packages/player/src/hooks/use-notes-cues.ts

```diff
@@ -10,28 +10,54 @@ import {cueCountFromTracks} from '../utils/cue-count-from-tracks'
  * @return {VTTCue[]}
  */
 export const useNotesCues = () => {
+  const [trackListUpdated, setTrackListUpdated] = React.useState(false)
   const [cues, setCues] = React.useState<VTTCue[]>([])
   const videoService = useVideo()
   const metadataTracks = useSelector(videoService, selectMetadataTracks)
   const textTracks = useSelector(videoService, selectTextTracks)
 
+  /**
+   * listen to add an remove track events on the {TextTrackList} so we can
+   * clear and update the cues accordingly
+   */
+  React.useEffect(() => {
+    function handleTrackChanges() {
+      setTrackListUpdated(true)
+    }
+    if (textTracks) {
+      textTracks.addEventListener('addtrack', handleTrackChanges)
+      textTracks.addEventListener('removetrack', handleTrackChanges)
+    }
+
+    return () => {
+      if (textTracks) {
+        textTracks.removeEventListener('addtrack', handleTrackChanges)
+        textTracks.removeEventListener('removetrack', handleTrackChanges)
+      }
+    }
+  }, [textTracks])
+
   /**
    * monitor the textTracks on the current video element and register
    * any metadata tracks that appear
    */
   React.useEffect(() => {
-    // {TextTrackList} objects aren't real arrays so you have to
-    // convert them accordingly
-    const tracks = Array.from(textTracks || []).filter((track) => {
-      return ['metadata'].includes(track.kind)
-    })
+    if (trackListUpdated) {
+      videoService.send({type: 'CLEAR_METADATA_TRACKS'})
+
+      // {TextTrackList} objects aren't real arrays so you have to
+      // convert them accordingly
+      const tracks = Array.from(textTracks || []).filter((track) => {
+        return ['metadata'].includes(track.kind)
+      })
 
-    tracks.forEach((track) => {
-      if (track.mode !== 'showing') {
```

Comment:
since we are clearing them above, the mode doesn't matter here. It was here to prevent adding multiple instances to the array.